### PR TITLE
Adds additional negative reencryption tests

### PR DESCRIPTION
**Pull request by iambmelt**
Impetus:
- https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1300836

Adds additional negative test for "wrong key" scenarios

### CODE REVIEW

**Comment by iambmelt** [comment]
Converting to draft to add an additional test...

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java [hunk 1]
```diff
@@ -499,4 +499,264 @@ public void onTaskCompleted(Void aVoid) {
...
+                mTestEncrypterDecrypter,
+                mTestEncrypterDecrypter,
+                new ISharedPrefsFileManagerReencrypter.ReencryptionParams(
+                        false,
+                        true,
+                        false
+                )
+        );
+
+        // Assert nothing was done
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:584]: That's not what was happening...

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java [hunk 2]
```diff
@@ -499,4 +499,264 @@ public void onTaskCompleted(Void aVoid) {
...
+
+    @Test
+    public void testMultipleInvocationsAbortsAndClears() throws Exception {
+        // Add the numbers 0-9 to the cache, encrypted format...
+        for (int ii = 0; ii < 10; ii++) {
+            try {
+                final String encryptedIntStr = mTestEncrypterDecrypter.encryptWithLegacyKey(String.valueOf(ii));
+                mTestCacheFile.putString(String.valueOf(ii), encryptedIntStr);
+            } catch (Exception e) {
+                // wont happen
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:718]: Assert.assertNull("Should not have an exception here", e); ?

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java (hunk 1 above)
Comment (iambmelt) [inline common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:584]:
```suggestion
        // Assert entries removed
```

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java [hunk 3]
```diff
@@ -499,4 +499,264 @@ public void onTaskCompleted(Void aVoid) {
...
+
+    @Test
+    public void testMultipleInvocationsAborts() throws Exception {
+        // Add the numbers 0-9 to the cache, encrypted format...
+        for (int ii = 0; ii < 10; ii++) {
+            try {
+                final String encryptedIntStr = mTestEncrypterDecrypter.encryptWithLegacyKey(String.valueOf(ii));
+                mTestCacheFile.putString(String.valueOf(ii), encryptedIntStr);
+            } catch (Exception e) {
+                // wont happen
```
Comment (AdamBJohnsonx) [inline common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:666]: fail?

**Review by AdamBJohnsonx — APPROVED** [review]

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java (hunk 1 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java]: ea702c7b41052c5c8d4b5428971ea10f1f059d7d

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java (hunk 3 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:666]: ca7edc80e6809dae09f2b1575f02ffc0568988d7

File: common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java (hunk 2 above)
Comment (iambmelt) [reply common/src/androidTest/java/com/microsoft/identity/common/DefaultSharedPrefsFileManagerReencrypterTest.java:718]: ca7edc80e6809dae09f2b1575f02ffc0568988d7

**Pull request merged**